### PR TITLE
Add AuditLogError alert runbook

### PR DESCRIPTION
**Pull request by JAORMX**
As suggested in https://github.com/openshift/cluster-kube-apiserver-operator/pull/1166

Signed-off-by: Juan Antonio Osorio Robles <jaosorior@redhat.com>

### CODE REVIEW

File: alerts/AuditLogError.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  running out of disk space.
+
+* A malicious actor could be tampering with the audit log files or directory
+  permissions
+
+* The API server might be encountering an unexpected error.
+
+## Impact
+
+When there are error writing audit logs, security events will not be logged
```
Comment (NautiluX) [inline alerts/AuditLogError.md:21]:
```suggestion
When there are errors writing audit logs, security events will not be logged
```

File: alerts/AuditLogError.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+From the `instance` label, it's possible to determine what node is hosting
+the aforementioned affected API Server.
+
+Verify that the relevant audit log file permissions are what's expected:
+Owned by the `root` user and with a mode of `0600`. Make sure as well that
+there aren't unexpected attributes for the files, such as immutability or append-only.
+
+Verify that the relevant audit log directory permissions are what's expected:
+Owned by the `root` user and with a mode of `0700`.
```
Comment (NautiluX) [inline alerts/AuditLogError.md:49]: This is to be done on the node?

**Review by NautiluX — COMMENTED** [review]
Thanks for submitting this runbook, I think a bit of clarification in the Diagnosis section would be helpful where what should be done, other than that it looks good!

**Review by paulfantom — COMMENTED** [review]
It would be good to have a `Mitigation` section. Especially since the alert related to this runbook is a critical-level one.

**Comment by JAORMX** [comment]
@paulfantom I left that out intentionally as I'm not really sure what to add there. The response will be very different depending on the organization and the compliance requirements. A FedRAMP moderate deployment might need to isolate the node and investigate. A DoD Impact Level 5 deployment would need to snapshot and shut down the system immediately. A more normal deployment might just need to investigate, and since the causes could be many, it's hard to say how to mitigate it.

Given the feedback added on the alert, I downgraded it to a `warning` rather than a `critical` alert though.

**Comment by JAORMX** [comment]
Pull-request updated, HEAD is now 5a63b753b77534e4845028e94bf0439e6367c566

**Comment by JAORMX** [comment]
@NautiluX thanks for the review! Addressed your comments.

**Comment by paulfantom** [comment]
> The response will be very different depending on the organization and the compliance requirements. A FedRAMP moderate deployment might need to isolate the node and investigate. A DoD Impact Level 5 deployment would need to snapshot and shut down the system immediately. A more normal deployment might just need to investigate, and since the causes could be many, it's hard to say how to mitigate it.

Then let's write exactly that as mitigation. We can specify various routes the alert can be mitigated depending on the environment and even saying `If you suspect tampering is happening, contact your incident response team.` is potentially a sort of mitigation step.

**Comment by JAORMX** [comment]
Pull-request updated, HEAD is now 5acf85081b75944e6742e45a9ced1c7db4ab99f6

**Comment by JAORMX** [comment]
@paulfantom updated

**Review by paulfantom — COMMENTED** [review]
/approve

**Pull request merged**